themes/robbyrussell/robbyrussell.theme.bash

```diff
@@ -18,8 +18,19 @@ function git_prompt_info() {
   echo -e "$SCM_PREFIX${bold_red}$SCM_BRANCH$SCM_STATE$SCM_SUFFIX"
 }
 
+function venv_prompt() {
+	python_venv=""
+	# Detect python venv
+	if [[ -n "${CONDA_DEFAULT_ENV}" ]]; then
+		python_venv="($PYTHON_VENV_CHAR${CONDA_DEFAULT_ENV}) "
+	elif [[ -n "${VIRTUAL_ENV}" ]]; then
+		python_venv="($PYTHON_VENV_CHAR$(basename "${VIRTUAL_ENV}")) "
+	fi
+	[[ -n "${python_venv}" ]] && echo "${python_venv}"
+}
```

Comment:
May I suggest this variation instead?

```suggestion
function venv_prompt() {
	local python_venv=""

	if [[ -n "${CONDA_DEFAULT_ENV:-}" ]]; then
		python_venv="${CONDA_DEFAULT_ENV}"
		PYTHON_VENV_CHAR=${CONDA_PYTHON_VENV_CHAR}
	elif [[ -n "${VIRTUAL_ENV:-}" ]]; then
		python_venv="${VIRTUAL_ENV##*/}"
	fi

	[[ -n "${python_venv}" ]] && echo "${PYTHON_VENV_CHAR}${python_venv}"
}
```

This doesn't assume that any variables are already defined, sets `python_venv` as a `local` parameter to avoid environment pollution, and avoids a subshell and external command invocation. (This version is nearly identical to the one from `powerline/powerline.base`.)

Reply:
honestly, maybe we should incorporate this pattern into the base theme, so all themes could use it instead of duplicating the code

Reply:
Can you point me to the base theme?

Reply:
Thank you for the suggestion. This function returns name of the virtualenv without parentheses. Since in some themes name of the virtualenv is shown in parentheses, what is the best practice? To manage parentheses at the prompt, or to return them from `venv_prompt` function?

Reply:
I would say leave the parentheses to the theme

Reply:
Pushed changes :relaxed:

Reply:
I've added `venv_prompt` to base theme and pushed, all tests passed. However, not sure if this will break other themes that are also using their local `venv_prompt` functions (e.g. powerline). Should the naming stay the same?